src/controllers/users.controller.js

```diff
@@ -0,0 +1,43 @@
+/* eslint-disable require-jsdoc */
+/* eslint-disable valid-jsdoc */
+import bcrypt from 'bcrypt';
+import jwt from 'jsonwebtoken';
+import dotenv from 'dotenv';
+import validateUser from '../helpers';
+import { users } from '../database/models';
+
+dotenv.config();
+class UserController {
+  /**
+    * User can be able to sign up
+    * @param {object} req
+    * @param {object} res
+    */
+  static async signUp(req, res) {
+    try {
+      const { error } = validateUser.validateSignup(req.body);
+      if (error) {
+        throw new Error(error.details[0].message);
+      }
+      const hashedPassword = await bcrypt.hash(req.body.password, 10);
+      const { firstName, lastName, email } = req.body;
+      const isUserExist = await users.findOne({ where: { email } });
+
+      if (isUserExist) return res.status(409).json({ status: 409, error: 'User already exist' });
+
+      const datas = await users.create({
+        firstName, lastName, email, password: hashedPassword
+      });
+      const token = jwt.sign({
```

Comment:
@Willy also I think JWT authentication should be a helper function since it is reusable. Also, this is violation of the separation of concerns principle

Reply:
The feedback implemented